server/mlabns/util/fqdn_rewrite.py

```diff
@@ -0,0 +1,102 @@
+"""Rewrites raw M-Lab FQDNs to apply post-processing or annotations."""
+
+import logging
+
+from mlabns.util import message
+
+
+def rewrite(fqdn, address_family, tool_id):
+    """Rewrites an FQDN to add necessary annotations and special-casing.
+
+    Performs the following rewrites on an FQDN:
+    * Adds a v4/v6 annotation if the client requested an explicit address family
+    * Applies a workaround to fix FQDNs for NDT-SSL queries.
+
+    Args:
+        fqdn: A tool FQDN with no address family specific annotation.
+        address_family: The address family for which to create the FQDN or None
+            to create an address family agnostic FQDN.
+        tool_id: Name of tool associated with the FQDN (e.g. 'ndt_ssl').
+
+    Returns:
+        FQDN after rewriting to apply all modifications to the raw FQDN.
+    """
+    rewritten_fqdn = _apply_af_awareness(fqdn, address_family)
+    rewritten_fqdn = _apply_ndt_ssl_workaround(rewritten_fqdn, tool_id)
```

Comment:
Would it be better to wrap this in the conditional if tool_id == "ndt_ssl" rather than passing non NDT-SSL tools to a function named _apply_ndt_ssl_workaround(), and then doing the conditional in the that function?  Adding the conditional here is higher up in the chain and I think more readable, and will not pass control of the code to a function that shouldn't be working on the data its given in any case.